--- a/src/declarativeimports/core/private/DefaultToolTip.qml
+++ b/src/declarativeimports/core/private/DefaultToolTip.qml
@@ -31,7 +31,7 @@
         anchors.centerIn: parent
         anchors.margins: Kirigami.Units.largeSpacing
 
-        spacing: Kirigami.Units.gridUnit
+        spacing: Kirigami.Units.largeSpacing
 
         Image {
             source: root.toolTip ? root.toolTip.image : ""
@@ -44,8 +44,8 @@
             source: root.toolTip ? root.toolTip.icon : ""
             Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
             visible: root.toolTip !== null && root.toolTip.icon !== "" && root.toolTip.image === "" && valid
-            Layout.preferredWidth: Kirigami.Units.iconSizes.medium
-            Layout.preferredHeight: Kirigami.Units.iconSizes.medium
+            Layout.preferredWidth: Kirigami.Units.iconSizes.smallMedium
+            Layout.preferredHeight: Kirigami.Units.iconSizes.smallMedium
         }
 
         ColumnLayout {
@@ -53,10 +53,11 @@
             spacing: 0
 
             Kirigami.Heading {
-                level: 3
+                level: 4
                 Layout.fillWidth: true
                 elide: Text.ElideRight
-                wrapMode: Text.Wrap
+                wrapMode: Text.NoWrap
+                maximumLineCount: 1
                 text: root.toolTip ? root.toolTip.mainText : ""
                 textFormat: Text.PlainText
                 visible: text !== ""
@@ -69,7 +70,7 @@
                 textFormat: root.toolTip ? root.toolTip.textFormat : Text.AutoText
                 opacity: 0.75
                 visible: text !== ""
-                maximumLineCount: 8
+                maximumLineCount: 2
             }
         }
     }
