--- a/applets/systemtray/qml/ExpanderArrow.qml
+++ b/applets/systemtray/qml/ExpanderArrow.qml
@@ -19,10 +19,28 @@
     property int iconSize: Kirigami.Units.iconSizes.smallMedium
     implicitWidth: iconSize
     implicitHeight: iconSize
+
+    readonly property string toolTipIconName: {
+        if (Plasmoid.location === PlasmaCore.Types.TopEdge) {
+            return systemTrayState.expanded ? "arrow-up-symbolic" : "arrow-down-symbolic";
+        } else if (Plasmoid.location === PlasmaCore.Types.LeftEdge) {
+            return systemTrayState.expanded ? "arrow-left-symbolic" : "arrow-right-symbolic";
+        } else if (Plasmoid.location === PlasmaCore.Types.RightEdge) {
+            return systemTrayState.expanded ? "arrow-right-symbolic" : "arrow-left-symbolic";
+        }
+        return systemTrayState.expanded ? "arrow-down-symbolic" : "arrow-up-symbolic";
+    }
+
+    icon: toolTipIconName
+    mainText: systemTrayState.expanded ? i18n("Close status panel") : i18n("Hidden icons")
+    subText: systemTrayState.expanded
+        ? ""
+        : i18np("%1 background item", "%1 background items", root.hiddenModel.count)
+
     activeFocusOnTab: true
 
-    Accessible.name: subText
-    Accessible.description: i18n("Show all the items in the system tray in a popup")
+    Accessible.name: mainText
+    Accessible.description: subText
     Accessible.role: Accessible.Button
     Accessible.onPressAction: systemTrayState.expanded = !systemTrayState.expanded
 
@@ -36,8 +54,6 @@
         }
     }
 
-    subText: systemTrayState.expanded ? i18n("Close popup") : i18n("Show hidden icons")
-
     property bool wasExpanded
 
     TapHandler {
--- a/applets/systemtray/qml/PlasmoidItem.qml
+++ b/applets/systemtray/qml/PlasmoidItem.qml
@@ -22,6 +22,7 @@
 
     itemId: applet?.Plasmoid.pluginName ?? ""
     mainText: applet && (inVisibleLayout || (applet.toolTipMainText && applet.toolTipMainText != text)) ? applet.toolTipMainText : ""
+    icon: applet?.Plasmoid.icon ?? ""
     subText: applet?.toolTipSubText ?? ""
     mainItem: applet?.toolTipItem ?? null
     textFormat: applet?.toolTipTextFormat ?? 0 /* Text.AutoText, the default value */
--- a/applets/systemtray/qml/StatusNotifierItem.qml
+++ b/applets/systemtray/qml/StatusNotifierItem.qml
@@ -18,6 +18,7 @@
     itemId: model?.Id ?? ""
     text: model?.Title || model?.ToolTipTitle || ""
     mainText: inVisibleLayout || (model?.ToolTipTitle !== "" && model?.ToolTipTitle !== text) ? model?.ToolTipTitle ?? "" : ""
+    icon: iconItem.source
     subText: model?.ToolTipSubTitle ?? ""
     textFormat: Text.AutoText
     active: inVisibleLayout || text != mainText || subText.length > 0
--- a/applets/systemtray/qml/CompactApplet.qml
+++ b/applets/systemtray/qml/CompactApplet.qml
@@ -17,6 +17,7 @@
     anchors.fill: parent
 
     mainText: plasmoidItem?.toolTipMainText ?? ""
+    icon: plasmoidItem?.Plasmoid.icon ?? ""
     subText: plasmoidItem?.toolTipSubText ?? ""
     location: if ((plasmoidItem?.parent as AbstractItem)?.inHiddenLayout && Plasmoid.location !== PlasmaCore.Types.LeftEdge) {
         return PlasmaCore.Types.RightEdge;
